app/views/tags/edit.html.erb

```diff
@@ -13,18 +13,17 @@
                  },
                  class: "app-c-contextual-guidance__form" do %>
       <% @edition.document_type.tags.each do |tag_field| %>
-        <% if @edition.access_limit.present? && tag_field.id == "primary_publishing_organisation" %>
-          <%= render "govuk_publishing_components/components/warning_text", {
-            text: t("tags.edit.organisation_warning"),
-          } %>
-        <% end %>
-
-        <div id="<%= tag_field.id %>">
-          <%= render "tags/tags/#{tag_field.type}_input",
+        <% template = if lookup_context.template_exists?(tag_field.id, "tags/tags", true)
+                        "tags/tags/#{tag_field.id}"
+                      else
+                        "tags/tags/default"
+                      end %>
+        <%= render template, edition: @edition, tag_field: tag_field do %>
+          <%= render "tags/inputs/#{tag_field.type}_input",
             tag_field: tag_field,
             tags: params[:tags] || @edition.tags,
             issues: @issues %>
-        </div>
+          <% end %>
```

Comment:
This does feel like a lot of duplication even with a partial, especially where things like `id` can be retrieved dynamically via `tag_field.id` rather than duplicating the ID in the model and the view. But I do like that we'd only have one view-rendering approach for all field types. I'll follow your suggestion and see what it looks like.